core/src/plan/validate.rs

```diff
@@ -0,0 +1,48 @@
+use {
+    super::PlanError,
+    crate::{
+        ast::{Expr, SelectItem, SetExpr, Statement},
+        data::Schema,
+        result::Result,
+    },
+    itertools::Itertools,
+    std::collections::HashMap,
+};
+
+/// Validate user select column should not be ambiguous
+pub fn validate(schema_map: &HashMap<String, Schema>, statement: Statement) -> Result<Statement> {
+    if let Statement::Query(query) = &statement {
+        if let SetExpr::Select(select) = &query.body {
+            if !select.from.joins.is_empty() {
+                select
+                    .projection
+                    .iter()
+                    .map(|select_item| {
+                        if let SelectItem::Expr {
+                            expr: Expr::Identifier(ident),
+                            ..
+                        } = select_item
+                        {
+                            let tables_with_given_col = schema_map
+                                .iter()
+                                .filter_map(|(_, schema)| {
+                                    schema.column_defs.iter().find(|col| &col.name == ident)
+                                })
+                                .collect_vec();
```

Comment:
<https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.count>

Not try using the iterator's count without having to make it a `Vec`?

Reply:
Aha you're right